dao/src/main/java/greencity/repository/LocationRepository.java

```diff
@@ -17,4 +17,13 @@
         + "JOIN BAG AS B ON OBM.ORDER_ID = :orderId and OBM.BAG_ID = B.ID ) "
         + "LIMIT 1 ", nativeQuery = true)
     Location findByOrderId(@Param("orderId") Long id);
+
+    /**
+     * {@inheritDoc}
+     */
+    @Query(nativeQuery = true,
+        value = "select * FROM locations as l "
+            + "join location_translations lt on l.id = lt.location_id "
+            + "where lt.location_name = :locationName")
+    Location findLocationByName(@Param("locationName") String locationName);
```

Comment:
if location is not found by name, you'll get null, right? Good practice is to use Optional as a return type instead.